perf/perfevents_24_7.py

```diff
@@ -0,0 +1,191 @@
+
+        # Check if missing parameter is reported
+        result1 = process.run(
+            'perf stat -v -C 0 -e hv_24x7/HPM_0THRD_NON_IDLE_CCYC/ sleep 1',
+            ignore_status=True)
+        if "invalid or unsupported event" not in result1.stderr or\
+                "Required parameter 'domain' not specified" not in\
+                result1.stdout:
+            nfail += 1
+        else:
+            self.log.info('perf recognized unsupported event')
+
+        # Check param=value format works
+        result2 = process.run('perf stat -v -C 0 -e\
+            hv_24x7/HPM_0THRD_NON_IDLE_CCYC,domain = 2,\
+            core = 1/ sleep 1', ignore_status=True)
+        if "Performance counter stats for 'CPU(s) 0'" not in result2.stderr:
+            nfail += 1
+        else:
+            self.log.info('perf recognized domain name in param=value format')
+
+        # Testing feature: Check using domain which is not existing
+        result2 = process.run(
+            'perf stat -v -C 0 -e hv_24x7/HPM_0THRD_NON_IDLE_CCYC,\
+            domain = 12,core = 1/ sleep 1', ignore_status=True)
+        if "not supported" not in result2.stderr:
+            nfail += 1
+        else:
+            self.log.info('perf listed non-existing domain as unsupported')
+
+        # Check for all domains
+        for domain in range(1, 6):
+            result_perf = process.run(
+                'perf stat -v -C 0 -e hv_24x7/HPM_0THRD_NON_IDLE_CCYC,\
+                domain = %s,core = 1/ sleep 1' % domain, ignore_status=True)
+            if "Performance counter stats for 'CPU(s) 0'" not in\
+                    result_perf.stderr:
+                nfail += 1
+            else:
+                self.log.info(
+                    'perf recognized domain name in param=value\
+                    format for all 6 domains')
+
+        # Check for all domains
+        # Testing feature: Fix usage with chip events
+
+        event_out = process.run("cat %s/PM_XLINK_CYCLES" % EVENT_SYSFS)
+        if "chip=?" in event_out.stdout:
+            self.log.info('sysfs entry has chip entry')
+        else:
+            nfail += 1
+
+        if os.path.exists("/sys/bus/event_source/devices/hv_24x7/format/chip"):
+            self.log.info('chip file exists')
+        else:
+            nfail += 1
+
+        # Check missing chip parameter
+        output_chip_missing = process.run(
+            'perf stat -C 0 -v -e hv_24x7/PM_XLINK_CYCLES,\
+            domain = 1/ /bin/true', ignore_status=True)
+        if "Required parameter 'chip' not specified"\
+                not in output_chip_missing.stdout:
+            nfail += 1
+        else:
+            self.log.info('perf detected chip parameter missing')
+
+        output_chip = process.run(
+            'perf stat -C 0 -v -e hv_24x7/PM_XLINK_CYCLES,\
+            domain = 1,chip = 1/ /bin/true')
+        if "Performance counter stats for" not in output_chip.stderr:
+            nfail += 1
+
+        if nfail != 0:
```

Comment:
Would be better to have more info, like, what all failed, instead of just failed count